registry/remote/repository.go

```diff
@@ -234,6 +236,28 @@ func (r *Repository) client() Client {
 	return r.Client
 }
 
+func (r *Repository) do(req *http.Request) (*http.Response, error) {
+	if r.HandleWarning == nil {
+		return r.client().Do(req)
+	}
+
+	resp, err := r.client().Do(req)
+	if err != nil {
+		return nil, err
+	}
+	warningHeaders := parseWarningHeaders(resp.Header[headerWarning])
```

Comment:
Headers are case insensitive. You may consider [`Header.Values()`](https://pkg.go.dev/net/http#Header.Values).

Reply:
Good catch!